File: backend/src/Domain/Entities/Relationship.cs
using backend.Domain.Enums;

namespace backend.Domain.Entities;

public class Relationship : BaseEntity
{
    public Guid SourceMemberId { get; set; }
    public Member SourceMember { get; set; } = null!;

    public Guid TargetMemberId { get; set; }
    public Member TargetMember { get; set; } = null!;

    public RelationshipType Type { get; set; }
    public int? Order { get; set; } // For birth order, etc.
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? Description { get; set; }
}

File: frontend/src/types/family/relationship.ts
import type { RelationshipType } from './relationship-type';

export interface RelationshipMember {
  dateOfBirth?: Date;
  fullName: string;
  gender?: string;
  avatarUrl?: string;
}

export interface Relationship {
  id: string;
  sourceMemberId: string;
  targetMemberId: string;
  type: RelationshipType;
  order?: number;
  sourceMember?: RelationshipMember;
  targetMember?: RelationshipMember;
}
